sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/DoNotUseCollectionInItsOwnMethodCalls.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Tests.Diagnostics
+{
+    class Program : IEqualityComparer<int>
+    {
+        public bool Equals(int left, int right) => left == right;
+
+        void InvalidCases()
+        {
+            var list = new List<int>();
+            var set = new HashSet<int>();
+
```

Comment:
Few more methods to track:
Set: IsProperSubsetOf, IsProperSupersetOf, IsSubsetOf, IsSupersetOf, Overlaps, SetEquals, SymmetricExceptWith
List: InsertRange